### PR TITLE
Fixes the HorizontalColourBar to show the roundedness correctly

### PR DESCRIPTION
**Pull request by sai-manoj-kumar**
Fix for the Issue #178.

| Before Fix  | After Fix |
| ------------- | ------------- |
|  before fix the corner radius| after fix the corner radius is correct |

### CODE REVIEW

**Comment by sai-manoj-kumar** [comment]
@willdale Can you please review this PR?

File: Sources/SwiftUICharts/BarChart/Shapes/RoundedRectangleBarShape.swift [hunk 1]
```diff
@@ -45,10 +45,10 @@ internal struct RoundedRectangleBarShape: Shape {
         let h = rect.size.height
         
         // Make sure we do not exceed the size of the rectangle
-        let tr = min(min(self.tr, h/2), w/2)
```
Comment (willdale) [inline Sources/SwiftUICharts/BarChart/Shapes/RoundedRectangleBarShape.swift:48]: As the comment above suggests, part of what is here is to control the size of the radius.

Screenshot 2022-04-02 at 08 04 54

Without the `/2` there can be drawing errors.

File: Sources/SwiftUICharts/BarChart/Views/SubViews/Bars.swift [hunk 2]
```diff
@@ -500,19 +500,25 @@ internal struct HorizontalColourBar<CD: CTBarChartDataProtocol & GetDataProtocol
...
-                self.startAnimation = true
-            }
-            .animateOnDisappear(using: chartData.chartStyle.globalAnimation) {
-                self.startAnimation = false
-            }
-            .accessibilityValue(dataPoint.getCellAccessibilityValue(specifier: chartData.infoView.touchSpecifier,
-                                                                    formatter: chartData.infoView.touchFormatter))
+        let scaleRatio = divideByZeroProtection(CGFloat.self, dataPoint.value, chartData.maxValue)
+        let minWidth = max(chartData.barStyle.cornerRadius.topLeft, chartData.barStyle.cornerRadius.bottomLeft) + max(chartData.barStyle.cornerRadius.topRight, chartData.barStyle.cornerRadius.bottomRight)
+        return GeometryReader { geo in
```
Comment (willdale) [inline Sources/SwiftUICharts/BarChart/Views/SubViews/Bars.swift:505]: My preference is to avoid having to put return statements into the view.

**Comment by sai-manoj-kumar** [comment]
@willdale Addressed your comments. Can you please review it again?

**Comment by willdale** [comment]
Failing build on iOS 14 and tvOS 14: Cannot convert value of type 'Double' to expected argument type 'CGFloat'

**Comment by willdale** [comment]
The chart no longer animates on appear.

**Comment by sai-manoj-kumar** [comment]
@willdale, I have fixed the animation issue and build error.

**Comment by willdale** [comment]
I'm going to decline this PR. The implementation is too specific to one chart type. I've got an implementation from V3 which I will back port for all bar chart types.

Thanks,
Will

**Pull request closed without merging**